tests/test_testing.py

```diff
@@ -0,0 +1,67 @@
+import math
+import time
+
+import pytest
+
+from asgiref.testing import ApplicationCommunicator
+from asgiref.wsgi import WsgiToAsgi
+
+
+@pytest.mark.parametrize(
+    "event, timeout, interval, duration, tolerance, expected", [
+        # No event
+        # Standard values
+        (False, 0.1, 0.01, 0.1, 0.03, True),
+        # interval > timeout (interval has precedence over timeout)
+        (False, 0.1, 1, 1, 0.03, True),
+        # timeout % interval > 0 (interval cycle completes)
+        (False, 1, 0.6, 1.2, 0.03, True),
+
+        # Events
+        # Standard values
+        (True, 0.1, 0.01, 0.01, 0.003, False),
+        # No delay
+        (True, 0, 0.01, 0, 0.003, True),
```

Comment:
I'm not sure whether to keep this test or not because it might fail in some environments (the message might be in the queue already).

Reply:
Don't keep it like this - there is no need to be this precise. All you need to do is test that it returns `True` or `False` correctly and within a simple overall timeout.

Reply:
I thought so. However, I wanted to make sure that the `reveice_nothing` behaves as I supposed in the first place. - I've updated the test.